wordpress-mobile/WordPress-iOS
# Jetpack App (Emphasis): Download App Store Strings

Part of #16395

## Description 
This PR adds the`download_jetpack_promo_strings` lane, which downloads the app store metadata strings from GlotPress 

Design ref: pcdRpT-n0-p2

## How to test

Download app store strings
1. Run `bundle exec fastlane download_jetpack_promo_strings`
2. ✅ Check that the strings are downloaded to `fastlane/appstoreres/jetpack_metadata`

Create Jetpack promo screenshots

0. You can download the Jetpack promo screenshots here (p1623662644138700-slack-C0180B5PRJ4), or go through the steps below to generate your own.
1. First we need to generate raw screenshots. If you already have raw screenshots for all four devices defined in `ScreenshotFastfile`, go ahead and skip to Step 5
2. Run`rake mocks`
3. Run `bundle exec fastlane jetpack_screenshots`
4. ✅ The raw screenshots are generated in the `fastlane/jetpack_screenshots` directory
5. ⚠️ The current release toolkit version returns an error `[!] Unable to determine the plugin root directory`. Here's a workaround: open `fastlane/Pluginfile` and point the release toolkit to `1.0.0` 
```
gem 'fastlane-plugin-wpmreleasetoolkit', git: 'https://github.com/wordpress-mobile/release-toolkit', tag: '1.0.0'
```
6. ⚠️ You also need to comment out this line in`fastlane/Fastfile`. We need to do this because the `check_for_toolkit_updates` action relies on release toolkit `1.2.0`, but we're setting the toolkit to `1.0.0` in the previous step
```
check_for_toolkit_updates unless is_ci || ENV['FASTLANE_SKIP_TOOLKIT_UPDATE_CHECK']
```
7. Run `bundle install`
8. Run `bundle exec fastlane create_jetpack_promo_screenshots`
9. ✅ The promo screenshots are generated in the `fastlane/jetpack_promo_screenshots` directory

## Blockers

1. ` create_jetpack_promo_screenshots` is failing when using the current release toolkit version (1.2.0). The error message is `[!] Unable to determine the plugin root directory`. Since this is a release toolkit issue, [this also affects Android](https://github.com/wordpress-mobile/WordPress-Android/pull/14744). I haven't done much digging, but I noticed ` create_jetpack_promo_screenshots` works if you revert the toolkit version to 1.0.0.
2. `jetpack_screenshots` is failing on iPad devices for Russian. It works fine for all other devices/language combos. I think this is happening because the simulator keyboard is Russian, but "typing" in the password in the automated test requires an English keyboard. I haven't been able to find a proper solution, but a hacky workaround is to quickly tap on the 'ABC' button on the Russian keyboard to switch to the English keyboard, while the UI test is running on the iPad simulator. I have no idea why this only happens on iPad devices for Russian. 😞 

## Regression Notes
1. Potential unintended areas of impact
WordPress fastlane logic

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Ran and checked the Wordpress screenshot lanes

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
